src/mono/wasm/Wasm.Build.Tests/TestAppScenarios/SignalRClientTests.cs

```diff
@@ -0,0 +1,117 @@
+    public async Task SignalRPassMessages(string config, string transport)
+    {
+        BlazorHostedBuild(config,
+            assetName: "BlazorHostedApp",
+            clientDirRelativeToProjectDir: "../BlazorHosted.Client",
+            generatedProjectNamePrefix: "SignalRClientTests",
+            runtimeType: RuntimeVariant.MultiThreaded);
+
+        List<string> consoleOutput = new();
+        List<string> serverOutput = new();
+
+        // We are using build (not publish),
+        // we need to instruct static web assets to use manifest file,
+        // because wwwroot in bin doesn't contain all files (for build)
+        s_buildEnv.EnvVars["ASPNETCORE_ENVIRONMENT"] = "Development";
+        using var runCommand = new RunCommand(s_buildEnv, _testOutput)
+                                    .WithWorkingDirectory(_projectDir!);
+        await using var runner = new BrowserRunner(_testOutput);
+        var url = await runner.StartServerAndGetUrlAsync(
+            cmd: runCommand,
+            args: $"run -c {config} --no-build",
+            onServerMessage: OnServerMessage);
+        var chatUrl = url + $"/chat?transport={transport}&message=ping";
+        IBrowser browser = await runner.SpawnBrowserAsync(url);
+        IBrowserContext context = await browser.NewContextAsync();
+
+        var page = await runner.RunAsync(context, chatUrl);
+        try
+        {
+            #pragma warning disable 4014
+            page.Console += (_, msg) => OnConsoleMessage(msg);
+            #pragma warning restore 4014
+            _testOutput.WriteLine($"Wait for loading state");
+            await page.WaitForLoadStateAsync(LoadState.DOMContentLoaded);
+            await page.ClickAsync("button#connectButton");
+            // give the connection some time to establish
+            await Task.Delay(3000);
+        }
+        catch (Exception ex)
+        {
+            Console.WriteLine($"Exception: {ex}");
+            Console.WriteLine($"ConsoleOutput: {string.Join(Environment.NewLine, consoleOutput)}");
+            Console.WriteLine($"ServerOutput: {string.Join(Environment.NewLine, serverOutput)}");
+            throw;
+        }
+
+        async Task OnConsoleMessage(IConsoleMessage msg)
+        {
+            consoleOutput.Add(msg.Text);
+            if (msg.Text.Contains("TestOutput ->"))
+            {
+                _testOutput.WriteLine(msg.Text);
+            }
+
+            if (msg.Text.Contains("SignalR connected"))
+                await page.ClickAsync("button#subscribeButton");
+
+            if (msg.Text.Contains("Subscribed to ReceiveMessage"))
+                await page.ClickAsync("button#sendMessageButton");
+
+            if (msg.Text.Contains("ReceiveMessage from server"))
+                await page.ClickAsync("button#exitProgramButton");
+
+            if (msg.Text.Contains("Exit signal was sent"))
+            {
+                await runner.WaitForExitMessageAsync(TimeSpan.FromSeconds(10));
```

Comment:
Since nobody is awating the method, this call will not block anything, am I correct?

Reply:
Correct, we could finish the test without noticing it did not return. Having this information we can remove it which makes it much easier to implement the clean-up. Awesome!